assets/app/scripts/services/apiEndpoint_v2.js

```diff
@@ -0,0 +1,311 @@
+'use strict';
+/* jshint unused:false */
+
+// see gist: https://gist.github.com/benjaminapetersen/540ac605d3c1b0660062
+// for discussion leading up to the creation of this file
+
+angular.module('openshiftConsole')
+  .factory('ApiEndpointService_v2', function(API_CFG) {
+
+    // defaultVersion lookup for backwards compatibility...
+    var defaultVersionByGroup = {
+      'extensions':   'v1beta1',
+      'experimental': 'v1beta1',
+      '':             'v1'  // this group applies to openshift & k8s (/oapi) & (/api)
+    };
+
+    // private processor
+    var mapApis = function(groupVersions) {
+      return _.reduce(
+              groupVersions,
+              function(memo, next) {
```

Comment:
Ok let me think on this.  I always do `memo` and `next` w/a `reduce` as a standard thing because `memo` is what you are building up and 'next' is the next thing to add (and memo=first the first time the fn is called, so this seems to make sense).  Its a nice pattern when every reduce looks the same, but I'm open to adjusting.